Framework/Beamline/inc/MantidBeamline/DetectorInfo.h

```diff
@@ -95,32 +94,27 @@ class MANTID_BEAMLINE_DLL DetectorInfo {
   void setRotation(const std::pair<size_t, size_t> &index,
                    const Eigen::Quaterniond &rotation);
 
-  size_t scanCount(const size_t index) const;
-  std::pair<int64_t, int64_t>
-  scanInterval(const std::pair<size_t, size_t> &index) const;
-  void setScanInterval(const size_t index,
-                       const std::pair<int64_t, int64_t> &interval);
+  size_t scanCount() const;
+  const std::vector<std::pair<int64_t, int64_t>> &scanIntervals() const;
   void setScanInterval(const std::pair<int64_t, int64_t> &interval);
```

Comment:
Is there any reason for having different intervals in `ComponentInfo` and `DetectorInfo`? I think not, so in that case there are two options:
1. One forwards to the other (intervals stored only in one place).
1. Completely remove one of them.

Neither seems like a great option, are there others?